clang/lib/Basic/Targets/NativeCPU.cpp

```diff
@@ -0,0 +1,109 @@
+//===--- NativeCPU.cpp - Implement NativeCPU target feature support -------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements NativeCPU TargetInfo objects.
+//
+//===----------------------------------------------------------------------===//
+
+#include "NativeCPU.h"
+#include <llvm/TargetParser/Host.h>
+
+using namespace clang;
+using namespace clang::targets;
+
+static const unsigned NativeCPUASMap[] = {
+    0,  // Default
+    1,  // opencl_global
+    3,  // opencl_local
+    4,  // opencl_constant
+    0,  // opencl_private
+    0,  // opencl_generic
+    1,  // opencl_global_device
+    1,  // opencl_global_host
+    1,  // cuda_device
+    4,  // cuda_constant
+    3,  // cuda_shared
+    1,  // sycl_global
+    1,  // sycl_global_device
+    1,  // sycl_global_host
+    3,  // sycl_local
+    0,  // sycl_private
+    0,  // ptr32_sptr
+    0,  // ptr32_uptr
+    0,  // ptr64
+    0,  // hlsl_groupshared
+    0,  // hlsl_constant
+    20, // wasm_funcref
+};
+
+NativeCPUTargetInfo::NativeCPUTargetInfo(const llvm::Triple &,
+                                         const TargetOptions &Opts)
+    : TargetInfo(llvm::Triple()) {
+  AddrSpaceMap = &NativeCPUASMap;
+  UseAddrSpaceMapMangling = true;
+  HasLegalHalfType = true;
+  HasFloat16 = true;
+  resetDataLayout("e");
```

Comment:
AFAIK little endian is specified by default. I wonder if we need to call `resetDataLayout` at all since we're not giving any more info.

Reply:
We do need that, yes, otherwise we get `Assertion `!DataLayoutString.empty() && "Uninitialized DataLayout!"' failed.`